src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +11,36 @@
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        Lock writeLock = lock.writeLock();
+        writeLock.lock();
+        try {
+            list.add(element);
+        } finally {
+            writeLock.unlock();
+        }
     }
 
     public E get(int index) {
-        // write your code here
-        return null;
+        E element = null;
+        Lock readLock = lock.readLock();
+        readLock.lock();
+        try {
+            element = list.get(index);
```

Comment:
in this case, it is better to return value directly from the try block
```suggestion
            return list.get(index);
```